### PR TITLE
Merge dev to main.

### PR DESCRIPTION
**Pull request by mahesh-panchal**
- Various bug fixes
- Integrates a single run of Hifiasm
- Integrates purge dups

The workflow runs through on a test data set.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -1,24 +1,45 @@
 #! /usr/bin/env nextflow
 
-nextflow.enable.dsl = 2
+include { PREPARE_INPUT } from "$projectDir/subworkflows/local/prepare_input/main"
 
-include { PREPARE_INPUT } from "$projectDir/subworkflows/prepare_input/main"
+include { BUILD_DATABASES as BUILD_HIFI_DATABASES } from "$projectDir/subworkflows/local/build_databases/main"
```
Comment (maxulysse) [inline main.nf:5]: I like the inclusion with `$projectDir`, I proposed that in nf-core, but got denied...

File: modules/local/mash/filter.nf [hunk 2]
```diff
@@ -2,7 +2,7 @@ process MASH_FILTER {
     tag "$meta.id"
     label 'process_low'
 
-    conda (params.enable_conda ? "bioconda::mash=2.3" : null)
+    conda "bioconda::mash=2.3"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/mash:2.3--he348c14_1':
         'quay.io/biocontainers/mash:2.3--he348c14_1' }"
```
Comment (maxulysse) [inline modules/local/mash/filter.nf:8]: you can get rid of the quay.io here

File: modules/local/samtools/fasta/main.nf [hunk 3]
```diff
@@ -2,7 +2,7 @@ process SAMTOOLS_FASTA {
     tag "$meta.id"
     label 'process_low'
 
-    conda (params.enable_conda ? "bioconda::samtools=1.14" : null)
+    conda "bioconda::samtools=1.14"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/samtools:1.14--hb421002_0' :
         'quay.io/biocontainers/samtools:1.14--hb421002_0' }"
```
Comment (maxulysse) [inline modules/local/samtools/fasta/main.nf:8]:
```suggestion
        'biocontainers/samtools:1.14--hb421002_0' }"
```

File: modules/local/mash/filter.nf (hunk 2 above)
Comment (maxulysse) [reply modules/local/mash/filter.nf:8]:
```suggestion
        'biocontainers/mash:2.3--he348c14_1' }"
```

File: modules/nf-core/mitohifi/findmitoreference/main.nf [hunk 4]
```diff
@@ -0,0 +1,47 @@
+process MITOHIFI_FINDMITOREFERENCE {
+    tag '$species'
+    label 'process_low'
+
+    // Docker image available at the biocontainers Dockerhub
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker://biocontainers/mitohifi:3.0.0_cv1':
+        'docker.io/biocontainers/mitohifi:3.0.0_cv1' }"
```
Comment (maxulysse) [inline modules/nf-core/mitohifi/findmitoreference/main.nf:8]: What's happening there?

File: modules/nf-core/purgedups/splitfa/purgedups-splitfa.diff [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+     task.ext.when == null || task.ext.when
+@@ -20,8 +21,10 @@
+     script:
+     def args = task.ext.args ?: ''
+     def prefix = task.ext.prefix ?: "${meta.id}"
++    def useGzip = !( assembly instanceof List ? assembly.every{ it.name.endsWith(".gz") } : assembly.name.endsWith(".gz") )
+     """
+-    split_fa $args $assembly | gzip -c > ${prefix}.split.fasta.gz
++    ${useGzip ? 'gzip -c' : 'cat'} $assembly > ${prefix}.merged.fasta.gz
++    split_fa $args ${prefix}.merged.fasta.gz | gzip -c > ${prefix}.split.fasta.gz
```
Comment (maxulysse) [inline modules/nf-core/purgedups/splitfa/purgedups-splitfa.diff:24]: Any plans to add that back to nf-core/modules?

File: modules/nf-core/quast/quast.diff [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+--- modules/nf-core/quast/main.nf
++++ modules/nf-core/quast/main.nf
+@@ -9,8 +9,8 @@
+ 
+     input:
+     tuple val(meta) , path(consensus)
+-    tuple val(meta2), path(fasta)
+-    tuple val(meta3), path(gff)
++    path(fasta)
++    path(gff)
```
Comment (maxulysse) [inline modules/nf-core/quast/quast.diff:11]: Let's have a discussion about that

File: nextflow.config [hunk 7]
```diff
@@ -44,56 +44,67 @@ params {
     // General module configuration
     multiqc_config = "$baseDir/configs/multiqc_conf.yaml"
 
-    // Conda
-    enable_conda = false
 }
 
+// Set container registry information
+docker.registry      = 'quay.io'
+singularity.registry = 'quay.io'
```
Comment (maxulysse) [inline nextflow.config:51]: you're missing apptainer, and podman:
https://github.com/nf-core/tools/blob/9ab896c140ee3fde816d250f3d8353aee5547d78/nf_core/pipeline-template/nextflow.config#L183-L189

**Review by maxulysse — APPROVED** [review]
Looking good, just some minor comments

File: modules/nf-core/purgedups/splitfa/purgedups-splitfa.diff (hunk 5 above)
Comment (mahesh-panchal) [reply modules/nf-core/purgedups/splitfa/purgedups-splitfa.diff:24]: Yes, eventually.

File: modules/nf-core/mitohifi/findmitoreference/main.nf (hunk 4 above)
Comment (mahesh-panchal) [reply modules/nf-core/mitohifi/findmitoreference/main.nf:8]: The module is outdated. I fixed it up: https://github.com/nf-core/modules/blob/master/modules/nf-core/mitohifi/findmitoreference/main.nf, just not in this code base yet.

File: nextflow.config (hunk 7 above)
Comment (mahesh-panchal) [reply nextflow.config:51]:
```suggestion
apptainer.registry   = 'quay.io'
docker.registry      = 'quay.io'
podman.registry      = 'quay.io'
singularity.registry = 'quay.io'
```

**Pull request merged**